### PR TITLE
[token] add partial withdraw with capability

### PR DESCRIPTION
**Pull request by areshand**
### Description
address the request on enabling partial withdraw token with capability.

This allows people list x fungible tokens and buyer can buy a smaller than x amount. The rest of the tokens can still be listed on sale until all the tokens are bought

### Test Plan

added unit test

### CODE REVIEW

File: aptos-move/framework/aptos-token/sources/token.move [hunk 1]
```diff
@@ -272,6 +275,14 @@ module aptos_token::token {
         expiration_sec: u64,
     }
 
+    /// capability to do partial withdraw with signer. The capability owner can withdraw up to amount token
+    struct PartialWithdrawCapability has drop, store {
```
Comment (CapCap) [inline aptos-move/framework/aptos-token/sources/token.move:279]: what happens if I create both a `WithdrawCapability` and a `PartialWithdrawCapability` for the same item?

**Review by 0xchloe — APPROVED** [review]

File: aptos-move/framework/aptos-token/sources/token.move (hunk 1 above)
Comment (lightmark) [reply aptos-move/framework/aptos-token/sources/token.move:279]: Yeah, looks weird. It seems the `WithdrawCapability` can include `PartialWithdrawCapability` and add a `withdraw_with_capability_v2` with an extra `amount` param. So why not just reuse `WithdrawCapability`?

Comment (lightmark) [reply aptos-move/framework/aptos-token/sources/token.move:279]: From the API and the fields they have. There is no reason to distinguish them IMO. allowing partial withdraw should be a subset of allowing full withdraw. I mean, if you allow ppl to withdraw in full, you should allow them to withdraw partially by default.

Comment (lightmark) [reply aptos-move/framework/aptos-token/sources/token.move:279]: They is real a `CapCap` PR. hahahahahahah

Comment (areshand) [reply aptos-move/framework/aptos-token/sources/token.move]: updated, remove the redundant struct and only keep the new function

**Review by lightmark — APPROVED** [review]

**Pull request merged**